### PR TITLE
Generate bazel files with sbt-bazel

### PR DESCRIPTION
**Pull request by beala-stripe**
Use [sbt-bazel](https://github.com/stripe/sbt-bazel) to generate bazel build files and a workspace. This allows rainier to be consumed as a source dependency by bazel projects.

## Testing
You can test the generated files by trying to build the project with bazel:

To build everything:
`bazel build //...`

To run the example:
`bazel run rainier-example:logNormal`

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,11 +1,34 @@
...
+  settings(unpublished).
+  settings(
+    bazelWorkspaceGenerate := true,
+    bazelBuildGenerate := false,
+    bazelCustomWorkspace :=
+      WorkspacePrelude +:
+        MavenBindings +:
+        BazelString(
+          """
+          |git_repository(
```
Comment (beala-stripe) [inline build.sbt:19]: Manually add the jar_jar bazel rule definitions for shading.

File: build.sbt [hunk 2]
```diff
@@ -164,7 +207,39 @@ lazy val shadedAsm = project.
...
+        |  'scala_library',
+        |  'scala_test'
+        |)
+        |
+        |load(
+        |    "@com_github_johnynek_bazel_jar_jar//:jar_jar.bzl",
+        |    "jar_jar"
+        |)
+        |
+        |jar_jar(
```
Comment (beala-stripe) [inline build.sbt:225]: These two calls to jar_jar do the shading. Each is a separate target that rainier-core depends on up above.

File: build.sbt [hunk 3]
```diff
@@ -118,7 +147,21 @@ lazy val rainierExample = project.
   ).
   settings(commonSettings).
   settings(evilPlotCrossSettings).
-  settings(unpublished)
+  settings(unpublished).
+  settings(bazelCustomBuild := BuildPrelude +: BuildTargets +: BazelString(
+    """
+      |scala_binary(
```
Comment (beala-stripe) [inline build.sbt:153]: Add a binary target for the example in the readme.

File: build.sbt [hunk 4]
```diff
@@ -62,7 +85,13 @@ lazy val rainierCore = project.
   in(file("rainier-core")).
   settings(name := "rainier-core").
   dependsOn(shadedAsm).
-  settings(commonSettings)
+  settings(commonSettings).
+  settings(
+    bazelRuleDeps := (Deps(Compile) - ScalaLib(Compile)) +
+      BazelDep("//.rainier-shaded-asm", "asmTreeShaded") +
```
Comment (beala-stripe) [inline build.sbt:91]: This manually adds the two shaded targets that are created below. See the `shadedAsm` project.

**Review by beala-stripe — COMMENTED** [review]
I've added some comments explaining the less straight forward parts of the config.

**Comment by avibryant** [comment]
This all LGTM. There are definitely some things that would be nice to have but I can leave those as issues on sbt-bazel.

**Pull request merged**